pythonProject1/main.py

```diff
@@ -1,36 +1,49 @@
 import math
-#def is_prime(number: int):
+TARGET_NUMBER = 13195
 
+def largest_prime_factor(subject: int)-> int:
+    minimal_possible_answer = round(subject / 2)
+    multiplier = minimal_possible_answer
+    for y in reversed(range(minimal_possible_answer)):
 
-subject = 600851475143
-solution = 9
-answer = 0
-is_primal = True
-primal_checker = 0
-while solution < subject:
-    primal_checker = solution - 1
+        for y in reversed(range(minimal_possible_answer)):
+            if minimal_possible_answer * multiplier == subject and is_prime(minimal_possible_answer):
+                return minimal_possible_answer
+            multiplier -= 1
+        minimal_possible_answer -= 1
```

Comment:
If i'm not mistaken both of them remain exactly the same throguht this loop.
Why are they both neccessary?